modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/accept/language/AcceptLanguageImpl.java

```diff
@@ -82,9 +81,7 @@ else if (acceptLanguage.equals("zh-Hant-TW")) {
 				});
 
 			if (ListUtil.isEmpty(locales)) {
-				throw new NotAcceptableException(
-					"No locales match the accepted languages: " +
-						acceptLanguage);
+				return Collections.singletonList(company.getLocale());
```

Comment:
I am also reviewing this code:

```
public Locale getPreferredLocale() {
	List<Locale> locales = getLocales();

	if (ListUtil.isNotEmpty(locales)) {
		return locales.get(0);
	}

	try {
		User user = _portal.initUser(_httpServletRequest);

		return user.getLocale();
	}
	catch (NoSuchUserException noSuchUserException) {
		throw new NotFoundException(
			"Unable to get preferred locale from nonexistent user",
			noSuchUserException);
	}
	catch (Exception exception) {
		throw new InternalServerErrorException(
			"Unable to get preferred locale: " + exception.getMessage(),
			exception);
	}
}
```

Does it make sense this code now?

Reply:
Makes sense! I will take a view at this